airbase/fetch.py

```diff
@@ -276,3 +280,38 @@ async def fetch() -> None:
             pass
 
     asyncio.run(fetch())
+
+
+def fetch_to_single_parquet(
+    urls: list[str],
+    path: Path,
+    *,
+    progress: bool = DEFAULT.progress,
+    raise_for_status: bool = DEFAULT.raise_for_status,
+    max_concurrent: int = DEFAULT.max_concurrent,
+) -> None:
+    """Request a list of url write out all responses into a single parquet file
+
+    :param urls: requested urls
+    :param path: parquet file for all combined responses
+    :param progress: show progress bar
+    :param raise_for_status: Raise exceptions if download links
+        return "bad" HTTP status codes. If False,
+        a :py:func:`warnings.warn` will be issued instead.
+    :param max_concurrent: maximum concurrent requests
+    """
+    with tempfile.TemporaryDirectory() as tempdir:
+        fetch_to_directory(
+            urls=urls,
+            root=Path(tempdir),
+            progress=progress,
+            raise_for_status=raise_for_status,
+            max_concurrent=max_concurrent
+        )
+        if progress:
+            print("Merging results into single file...", file=sys.stderr)
+        ds = dataset.dataset(tempdir, format="parquet")
+        merged_dir = Path(tempdir) / "merged"
+        merged_file = merged_dir / "part-0.parquet"
+        dataset.write_dataset(ds, base_dir=merged_dir)
+        shutil.move(str(merged_file), str(path))
```

Comment:
I probably do not understand what is going on here.
Can the `TemporaryDirectory` be replaced with a context manager that renames the directory name on exit?

In most of my projects I use something like the following context manager to write into a temporary file name and rename or delete the file on exit. Maybe it can be adapted for your use?

``` python
@contextmanager
def atomic_path(path: Path, *, unique: bool = False) -> Iterator[Path]:
    """atomic write helper"""

    if unique:
        tmp = path.with_name(uuid4().hex)
        logger.debug(f"unique tmp path {tmp.name}")
    else:
        tmp = path.with_suffix(".tmp")

    try:
        yield tmp
        if tmp.exists():
            tmp.rename(path)
    finally:
        if tmp.exists():
            tmp.unlink()
```

Reply:
Also, I'm not sure if creating a multi-file parquet directory is still necessary. All the libraries that support parquet that I have used have a simple mechanism to concatenate dataframes.